### PR TITLE
Refactor documentation generation. Generate documentation for 0.28.0

### PR DESCRIPTION
**Pull request by customcommander**
I spent a few hours trying to get my head around the existing build system and thought that things could be simplified.

The whole build system was a little bit disjointed:

1. Starts with Make
2. Then finish off with a couple of disconnected NPM run scripts
3. The index file for the REPL seemed to have gone AWOL

Since we have `Make` already I consolidated everything with just a few Make targets. The REPL now shows the correct version and has the Gitter integration working.

I couldn't see why keeping the docs for all versions was necessary. Unless I am mistaken there is no way to access the docs for previous versions? I deleted lots of scripts and workflows related to keeping all the docs for all the versions. I didn't go as far as removing the docs themselves.

A couple of handlebars templates didn't seem to be in use anymore so I deleted them.

I also deleted lots of scripts and resources related to maintaining an index of releases. This seems to have been abandoned since 0.22.0 I think.

I took the liberty to disable the GitBook integration as it was failing the original build system anyway. I didn't delete any pages but since most of them are WIP/TODO I thought it was ok.

Finally I regenerated the documentation for 0.27.1 ;)

---

These issues https://github.com/ramda/ramda/issues/2716 and https://github.com/ramda/ramda/issues/2803 should be fixed. Possibly a few others too but I haven't checked yet.

Screenshot 2021-12-28 at 02 09 05

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,81 +1,17 @@
...
-index.html: check-version
-	echo '<!DOCTYPE html><html><head><link rel="canonical" href="http://ramdajs.com/$(VERSION)/index.html" /><script>window.location = "$(VERSION)/index.html" + window.location.hash;</script></head><body></body></html>' >'$@'
-
-%: $(VERSION)/%
-	cp '$<' '$@'
-
+all: docs/dist/ramda.js docs/index.html repl/index.html style.css index.html
 
-.PHONY: check-version
-check-version:
```
Comment (customcommander) [inline Makefile:66]: Not necessary anymore as it can be worked out from the package.json file

File: Makefile [hunk 2]
```diff
@@ -1,81 +1,17 @@
...
-	$(VERSION)/docs/dist/ramda.js \
-	$(VERSION)/docs/index.html \
-	$(VERSION)/docs/main.js \
-	$(VERSION)/index.html \
-	$(VERSION)/fonts/glyphicons-halflings-regular.eot \
-	$(VERSION)/fonts/glyphicons-halflings-regular.svg \
-	$(VERSION)/fonts/glyphicons-halflings-regular.ttf \
-	$(VERSION)/fonts/glyphicons-halflings-regular.woff \
-	$(VERSION)/fonts/glyphicons-halflings-regular.woff2 \
-	$(VERSION)/style.css \
```
Comment (customcommander) [inline Makefile:23]: Couldn't find a reason to keep all the documentations for all versions. We can use Git for that I think. Removing that allowed me to greatly simplify the entire Makefile and make the documentation generation process a bit simpler to follow.

File: make_index_html.js [hunk 3]
```diff
@@ -23,5 +21,5 @@ get_ramda_file('README.md')
     version: version
   })
 
-  fs.writeFileSync('index.html', html, {encoding: 'utf8'})
+  console.log(html)
```
Comment (customcommander) [inline make_index_html.js:24]: So that the output can be redirected to a file from a make target.

File: repl/index.html [hunk 4]
```diff
@@ -1,149 +1,8 @@
...
...nitial-scale=1" name="viewport"><title>Ramda Documentation</title><link href="../style.css" rel="stylesheet" type="text/css"><link href="https://cdn.rawgit.com/ramda/repl/1.2.0/dist/bundle.css" rel="stylesheet"><script src="https://cdnjs.cloudflare.com/ajax/libs/URI.js/1.18.1/URI.min.js"></script><script src="https://cdn.rawgit.com/ramda/repl/1.2.0/dist/bundle.js"></script><script src="index.js"></script></head><body><input id="open-nav" type="checkbox"><header class="navbar navbar-fixed-top navbar-inverse container-fluid"><div class="container-fluid"><div class="navbar-header"><label class="open-nav" for="open-nav"></label><a class="navbar-brand" href="#"><strong>Ramda</strong><span class="version"> v0.27.1</span></a></div><ul class="nav navbar-nav navbar-left"><li><a href="../">Home</a></li><li><a href="../docs/">Documentation</a></li><li class="active"><a href="../repl/">Try Ramda</a></li></ul><ul class="nav navbar-nav navbar-right"><li><a href="https://github.com/ramda/ramda">GitHub</a></li><li><a href="https://gitter.im/ramda/ramda">Discuss</a></li></ul></div></header><div class="ramda-repl-wrapper"><textarea class="ramda-repl-target"></textarea></div><script>window.gitter = {
```
Comment (customcommander) [inline repl/index.html:1]: Generated from the PUG template as opposed to from the Handlebars template which seems to have been deprecated. Gitter integration is working.

**Comment by CrossEye** [comment]
Somehow, I entirely missed this thread.  I'm merging now and will then try to run to generate 0.28.

File: Makefile (hunk 2 above)
Comment (CrossEye) [reply Makefile:23]: Yeah, they were a good idea at the start, but we've never really used them.

File: Makefile (hunk 1 above)
Comment (CrossEye) [reply Makefile:66]: I'm glad someone understands `make` besides David.  I never got my head around it.

**Review by CrossEye — COMMENTED** [review]
Post 1.0, I would like to look at switching from `make` to use GitHub actions instead.  This is great, though.

File: Makefile (hunk 1 above)
Comment (customcommander) [reply Makefile:66]: I'm actually quite a big fan of `make` so that didn't scare me :)

**Pull request merged**